.openshift-ci/get_latest_helm_chart_versions.py

```diff
@@ -0,0 +1,150 @@
+update_repo_cmd = "helm repo update"
+search_cmd = f"helm search repo {helm_repo_name} --versions --output json"
+remove_repo_cmd = f"helm repo remove {helm_repo_name}"
+
+Version = namedtuple("Version", ["major", "minor", "patch"])
+
+# Here we call "release" (or Y-Stream) the first appearance of X.Y.0 version.
+Release = namedtuple("Release", ["major", "minor"])
+
+# Default value of N, the number of previous releases to look up.
+# The current release cadence is 9 weeks (sometimes extended but not reduced), i.e. 9*7=63 days.
+# The current support period is 6 months, i.e. at most 184 days.
+# Therefore, at most 3 releases will be in support at any given moment of time with the current cadence and support
+# period.
+num_releases_default = 3
+
+
+def main(argv):
+    logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+    n = int(argv[1]) if len(argv) > 1 else num_releases_default
+    helm_versions = get_latest_helm_chart_versions("stackrox-secured-cluster-services", n)
+    logging.info(f"Helm chart versions for the latest {n} releases:")
+    print("\n".join(helm_versions))
+
+
+def get_latest_helm_chart_versions(chart_name, num_releases=num_releases_default):
+    add_helm_repo()
+    try:
+        update_helm_repo()
+        return __get_latest_helm_chart_versions(chart_name, num_releases)
+    finally:
+        remove_helm_repo()
+
+
+def __get_latest_helm_chart_versions(chart_name, num_releases):
+    charts = read_charts()
+    logging.info(f"Discovered total {len(charts)} charts")
+
+    filtered_charts = filter_charts_by_name(charts, chart_name)
+    logging.info(f"Found {len(filtered_charts)} charts with the given name {chart_name}")
+
+    latest_charts = get_latest_chart_for_each_release(filtered_charts)[:num_releases]
+    logging.debug(f"Identified these charts as {num_releases} latest: {latest_charts}")
+
+    return [c["version"] for c in latest_charts]
+
+
+def read_charts():
+    json_str = run_command(search_cmd, log_stdout=False)
+    charts_from_json = json.loads(json_str)
+
+    release_charts = [c for c in charts_from_json if is_release_version(c["app_version"])]
+
+    for entry in release_charts:
+        entry["parsed_app_version"] = parse_version(entry["app_version"])
+
+    return release_charts
+
+
+def is_release_version(version):
+    return re.search(r"^\d+\.\d+\.\d+$", version) is not None
+
+
+def parse_version(version_str):
+    nums = [int(s) for s in version_str.split(".")]
+    return Version(major=nums[0], minor=nums[1], patch=nums[2])
+
+
+def filter_charts_by_name(charts, chart_name):
+    return [c for c in charts if c["name"] == f"{helm_repo_name}/{chart_name}"]
+
+
+def get_latest_chart_for_each_release(charts):
+    sorted_charts = sorted(charts, key=lambda x: x["parsed_app_version"], reverse=True)
+
+    result = []
+    release = None
+
+    for chart in sorted_charts:
+        chart_release = version_to_release(chart["parsed_app_version"])
```

Comment:
A bit pedantic, but I don't think the `Release` version object has the correct mapping to `major` and `minor`. If you're using `app_version` here, you'll be parsing (for example) `74.3.0`, `major` will be `74` and `minor` will be `3`. But these are actually `minor` and `patch` versions.

Doesn't make much difference in the end result, but I was a bit confusing reading this.

It makes it even more confusing because technically for `4.0` we have major *and* minor in the same number (400).

No suggestions on how this could be improved.

Reply:
I was confused about the new numbering, I though `4.1` would be `401` on the chart. Thanks for clarifying. 

But still, versions `<4` have a different number position on the chart version, right? 

After 4.0: `400.1.23` => 400 = major; 1 = minor; 23 = patch
Before 4.0: `74.3.0` => 74 = minor; 3 = patch; 0 = ??